test/unit/specs/controllers/password-controller-test.js

```diff
@@ -0,0 +1,129 @@
+    scope.data = {password: 'testpassword', passwordConfirmation: 'test33'}
+    scope.status = {passwordConfirmValid: false};
+    expect(scope.confirmPasswordClass()).to.equal('glyphicon-remove');
+  });
+  
+  it('If the password and its confirmation partially match, show a glyhicon-none', function () {
+    scope.data = {password: 'testpassword', passwordConfirmation: 'test'}
+    scope.status = {passwordConfirmValid: false};
+    expect(scope.confirmPasswordClass()).to.equal('glyphicon-none');
+  });
+  
+  it('If the password is empty, passwordStrength should return an empty string', function () {
+    scope.data = {password: ''}
+    expect(scope.passwordStrength()).to.equal('');
+  });
+  
+  it('If the password is weak, passwordStrength should return WEAK', function () {
+    scope.data = {password: 'abc'}
+    expect(scope.passwordStrength()).to.equal('WEAK');
+  });
+  
+  it('If the password is not quite good enough, passwordStrength should return ALMOST', function () {
+    scope.data = {password: 'serend'}
+    expect(scope.passwordStrength()).to.equal('ALMOST');
+  });
+  
+  it('If the password is OK, passwordStrength should return GOOD', function () {
+    scope.data = {password: 'serendip'}
+    expect(scope.passwordStrength()).to.equal('GOOD');
+  });
+  
+  it('If the password is really good, passwordStrength should return STRONG', function () {
+    scope.data = {password: 'yAs5woN8E5oG5BA$wk'}
+    expect(scope.passwordStrength()).to.equal('STRONG');
+  });
+  
+  it('Only good passwords that comes with a matching confirmation should be valid', function () {
+    scope.data = {password: 'yAs5woN8E5oG5BA$wk', passwordConfirmation: 'yAs5woN8E5oG5BA$wk'}
+    scope.checkPassword();
+    expect(scope.status.passwordValid).to.be.true;
+    expect(scope.status.passwordConfirmValid).to.be.true;
+    expect(scope.errors.passwordErrors).to.be.empty;
+    expect(scope.errors.passwordConfirmErrors).to.be.empty;
+  });
+  
+  it('A weak password, even with a matching confirmation, should not be valid', function () {
+    scope.data = {password: 'hello', passwordConfirmation: 'hello'}
+    scope.checkPassword();
+    expect(scope.status.passwordValid).to.be.false;
+    expect(scope.status.passwordConfirmValid).to.be.true;
+    expect(scope.errors.passwordErrors).to.be.empty;
+    expect(scope.errors.passwordConfirmErrors).to.be.empty;
+  });
+  
+  it('Without a matching confirmation, a password should not be valid', function () {
+    scope.data = {password: 'yAs5woN8E5oG5BA$wk', passwordConfirmation: ''}
+    scope.checkPassword();
+    expect(scope.status.passwordValid).to.be.true;
+    expect(scope.status.passwordConfirmValid).to.be.false;
+    expect(scope.errors.passwordErrors).to.be.empty;
+    expect(scope.errors.passwordConfirmErrors).to.be.empty;
+  });
+  
+  it('If there is no password, validateInput should mark that as an error', function () {
+    scope.data = {password: ''}
+    var validInput = scope.validators.pop()();
```

Comment:
You should check all validators here. Right now there is only one but if more will be added this test may break.